src/pudl/output/eia860.py

```diff
@@ -322,6 +325,92 @@ def generators_eia860(pudl_engine, start_date=None, end_date=None):
+        the same reported source code.
+
+        As a secondary measure, this function also creates a dictionary for source codes
+        that have a one-to-one mapping to a technology description. After mapping the
+        generator specific values, these are also mapped to fill in more gaps.
+
+        """
+        # Keep track of len
+        len1 = len(df)
+        na_len = len(df['technology_description'].isna())
+        older_than_2013 = df['report_date'].dt.year > 2013
+        notna = df['technology_description'].notna()
+        idx = ['plant_id_eia', 'generator_id', 'energy_source_code_1']
+
+        # Create a dictionary that maps plant id, gen id and fuel code to a technology
+        # description. By sorting values from oldest to newest and dropping duplicates,
+        # we only keep the oldest associates for a given fuel code.
+        map_dict = (
+            df.sort_values(['report_date'])
+            .loc[older_than_2013 & notna]
+            .drop_duplicates(subset=idx)
+            .set_index(idx)
+            ['technology_description'].to_dict()
+        )
+
+        # For values that are NA, map tech descs specific to that plant generator. Note
+        # that you CANNOT define this isna above and use it twice after altering the
+        # amount of NA values because it will preserve the original slice and mess
+        # things up.
+        isna = df['technology_description'].isna()
+        df.loc[isna, 'technology_description'] = (
+            df[idx].agg(tuple, 1).map(map_dict)
+        )
+
+        # Make sure this NAs get filled (and more aren't accidentally created)
+        assert (new_na_len := len(df[df['technology_description'].isna()])) < na_len, \
+            'Supposed to fill tech desc NA.'
+        na_len = new_na_len
+
+        # Make dictionary of energy source codes that only ever map to one
+        # known technology description
+        static_fuels = (
+            df.dropna(subset=['technology_description'])
+            .drop_duplicates(subset=['energy_source_code_1', 'technology_description'])
+            .drop_duplicates(subset=['energy_source_code_1'], keep=False)
+            .set_index('energy_source_code_1')
+            ['technology_description'].to_dict()
+        )
+
+        # For values that are NA, map tech descs that are static throughout all the data
+        isna = df['technology_description'].isna()
+        df.loc[isna, 'technology_description'] = (
+            df.energy_source_code_1.map(static_fuels)
+        )
+
+        # Make sure this NAs get filled (and more aren't accidentally created)
+        assert (new_na_len := len(df[df['technology_description'].isna()])) < na_len, \
+            'Supposed to fill tech desc NA.'
+        na_len = new_na_len
+
+        # Make sure didn't change group length
+        assert len(df) == len1, 'group length getting altered'
+
+    if backfill_tech:
+        _backfill_tech_desc(out_df)
```

Comment:
It's not getting saved in it's pre-backfilled state if that's what you mean. If you run `pudl_out.gens_eia860(backfill_tech_desc=True)` it will have to reload the table (not cached) if you then run `pudl_out.gens_eia860()` because the default for `backfil_tech_desc` is False. If you run `pudl_out.gens_eia860(backfill_tech_desc=True)` twice or more in a row it will retrieve the cached version of the df with backfilled tech descriptions. 

I supposed I could try and figure out a way to also cache the out_df before it gets backfilled if that's what you mean?

Reply:
Oh no, I mean it doesn't look like you are returning the backfilled df here (I had expected to see `out_df = _backfill_tech_desc(out_df)`)

Reply:
ok sure I can do that!